docs/architecture/event-flows-proposal.md

```diff
@@ -0,0 +1,92 @@
+# Eventing flow proposal
+
+## Overview
+
+This document is a proposal for eventing flows we want to support with use of the [Management Plane](./terminology.md#management-plane)
+
+## Eventing with Kyma internal Event Bus
+
+![Management Plane Components](assets/internal-event-flow.svg)
+
+1. The event is registered using **Management Plane** API.
+2. The **Director** sends an event url to **EventGateway**.
+3. The **Agent** fetches the event data and exposes it inside **Runtime**.  
+4. The **Application** publishes an event.
+
+>**NOTE** The Event Gateway will support multiple protocols like HTTP or MQTT which can be specified inside EventSpec data.
+It will also support many authentication (e.g. OpenID Connect) and authorization (e.g. OAuth2) methods.
+## Eventing with external Event Bus
+
+![Management Plane Components](assets/external-event-flow.svg)
+
+1. The event is registered using **Management Plane** API with external event bus data.
+2. The **Agent** fetches the event and exposes it inside **Runtime**.
+3. **Eventing system** subscribes on external event bus.
+4. The application publishes an event.
+
+## GraphQL Schema  
+
+### Types
+```graphql
+
+scalar EventSubscriptionAttributes # -> map[string]string
+
+type EventAPIDefinition {
+    id: ID!
+    """group allows you to find the same API but in different version"""
+    group: String
+    spec: EventSpec!
+    auth: Auth
+    externalSubscriptions: [EventSubscription!]!
+    version: Version
+}
+
+type EventSpec {
+    data: CLOB
+    type: EventSpecType!
+    format: SpecFormat
+    fetchRequest: FetchRequest
+}
+
+type EventSubscription {
+    url: String!
+    auth: Auth
+    topic: String!
+    attributes: EventSubscriptionAttributes
+}
+
+
+enum EventSpecType {
+    ASYNC_API
+}
+
+```
+
+### Input Types
+
+```graphql
+
+input EventAPIDefinitionInput {
+    spec: EventSpecInput!
+    group: String
+    protocol: Protocol
```

Comment:
why input has protocol and EventAPIDefinition not?

Reply:
I've forgotten to remove that one, fixed already. 
To be clear about why I gave up on the idea of having Protocol field:
Inside AsyncAPI spec, there is a `Server Object` which has `schema` field. That field is basically specifying the protocol.
https://github.com/asyncapi/asyncapi/blob/master/versions/1.2.0/asyncapi.md#serverObject